### PR TITLE
Subpixel positioning for popup

### PR DESCRIPTION
**Pull request by ericmagnuson**
Hey, everyone!

This PR implements the idea [in this Discussion](https://github.com/maplibre/maplibre-gl-js/discussions/3614), allowing subpixel positioning of a `Popup`.  This addition allows a `Popup` to be animated more smoothly. The behavior is optional and can be set at instantiation or via class methods.  Please see the discussion for screen recordings of the behavior before and after.

### Details
A `subpixelPositioning` option (a boolean) is added to the `Popup`.  If the boolean is true, the offset rounding that happens when setting the CSS `translate` property of the `Popup` is bypassed.  The option can also be configured using a getter method, `Popup#setSubpixelPositioning`.

### Before
https://github.com/maplibre/maplibre-gl-js/assets/479605/369b3def-ceb4-4983-b493-2b2c94f847a4

### After
https://github.com/maplibre/maplibre-gl-js/assets/479605/cee14f85-3062-4b37-938b-c558360dc81f

## Launch Checklist

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality.
 - [x] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

File: src/ui/popup.ts [hunk 1]
```diff
@@ -523,6 +530,14 @@ export class Popup extends Evented {
         }
     }
 
+    enableSubpixelPositioning() {
```
Comment (HarelM) [inline src/ui/popup.ts:533]: Wouldn't it be easier to have a single `setSubpixelPositioning(value: bookean)`
as API?

**Comment by HarelM** [comment]
Looks good in general, thanks!
Can you add a before and after video by any chance?

File: src/ui/popup.ts (hunk 1 above)
Comment (ericmagnuson) [reply src/ui/popup.ts:533]: I debated between `toggle` and `disable`/`enable`, but I think you're totally right — `set` is even better! I'll push an update.

**Comment by ericmagnuson** [comment]
> Looks good in general, thanks! Can you add a before and after video by any chance?

There are before and after videos in the attached discussion — I'll pull the examples into the PR so that they are more readily available.

**Review by HarelM — APPROVED** [review]

**Comment by birkskyum** [comment]
So much nicer to look at, thanks!

**Pull request merged**